tools/github-actions/yarn-errors-reporter/src/main.ts

```
@@ -0,0 +1,71 @@
+import * as fs from 'node:fs';
+import * as os from 'node:os';
+import * as path from 'node:path';
+import * as core from '@actions/core';
+import {
+  exec,
+  getExecOutput,
+} from '@actions/exec';
+
+type YarnInstallOutputLine = { displayName: string; indent: string; data: string };
+
+function parseYarnInstallOutput(output: string, errorCodesToReport: string[]) {
+  return output.split(os.EOL)
+    .map((line) => line ? JSON.parse(line) as YarnInstallOutputLine : undefined)
+    .filter((line): line is YarnInstallOutputLine => !!line && errorCodesToReport.includes(line.displayName))
+    .map((line) => `➤${line.displayName}: ${line.indent}${line.data}`);
+}
+
+async function run(): Promise<void> {
+  try {
+    const cwd = process.env.GITHUB_WORKSPACE!;
+    const reportOnFile = core.getInput('reportOnFile');
+    const errorCodesToReport = core.getInput('errorCodesToReport').split(',');
+    const onlyReportsIfAffected = core.getBooleanInput('onlyReportsIfAffected');
+    const yarnLockPath = path.resolve(cwd, 'yarn.lock');
+    const execOptions = {
+      cwd,
+      ignoreReturnCode: true
+    };
+
+    const packageManager = fs.existsSync(yarnLockPath) ? 'yarn' : 'npm';
+    if (packageManager !== 'yarn') {
+      core.warning('This action only manages yarn, it doesn\'t do anything with other package managers');
+      return;
```

don't we want to return an error if it;s not used with the good package manager ?